interpreter/exec/v128.ml

```diff
@@ -1,5 +1,30 @@
-include Vector.Make
+include Simd.Make
   (struct
     include Bytes
     let bytewidth = 16
+
+    let of_strings shape ss =
+      let range_check i32 min max at =
+        let i = Int32.to_int i32 in
+        if i > max || i < min then raise (Failure "constant out of range") else i in
+      (* TODO create proper I8 and I16 modules *)
+      let i8_of_string s = range_check (I32.of_string s) (-128) 255 s in
+      let i16_of_string s = range_check (I32.of_string s) (-32768) 65535 s in
+      let open Bytes in
+      let b = create bytewidth in
+      (match shape with
+      | Simd.I8x16 ->
+        List.iteri (fun i s -> set_uint8 b i (i8_of_string s)) ss
+      | Simd.I16x8 ->
+        List.iteri (fun i s -> set_uint16_le b i (i16_of_string s)) ss
+      | Simd.I32x4 ->
+        List.iteri (fun i s -> set_int32_le b i (I32.of_string s)) ss
+      | Simd.I64x2 ->
+        List.iteri (fun i s -> set_int64_le b i (I64.of_string s)) ss
+      | Simd.F32x4 ->
+        List.iteri (fun i s -> set_int32_le b i (F32.to_bits (F32.of_string s))) ss
+      | Simd.F64x2 ->
+        List.iteri (fun i s -> set_int64_le b i (F64.to_bits (F64.of_string s))) ss);
+      if List.length ss != (Simd.lanes shape) then raise (Invalid_argument "wrong length")
```

Comment:
Good point, I put the length check after the constants parsing because the error messages in the test indicates that the ordering should be like so.
I.e. (v128.const f64x2 nan:0x0) will throw a constant out of range, rather than a unexpected token.

I wonder if we want to change the test assertions, or I can create a bytes of the right size, then truncate it at the end.

Reply:
The existing test suite tries to avoid prescribing any specific choice of diagnostic in the case where a program has multiple errors, simply by making sure that every negative test contains only one error. So I would propose changing the tests that way, such that the order of the checks does no longer matter.

Reply:
And you probably want to add a test that has too many literals.

Reply:
In the latest commit I removed these 3 of these checks, they looked redundant to the existing i32x4 checks. I moved 1 down and checked it to assert "unexpected token". Also added a test with too many literals.